### PR TITLE
IKASAN-1565 - Added support for emailProducer builder to fluentAPI.

### PR DESCRIPTION
**Pull request by mitcje**

### CODE REVIEW

File: ikasaneip/builder/jar/src/main/java/org/ikasan/builder/component/endpoint/EmailProducerBuilderImpl.java [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+        this.emailProducer.setConfiguration(emailProducerConfiguration);
+        return this;
+    }
+
+    @Override
+    public EmailProducerBuilder setToRecipient(String recipient)
+    {
+        List<String> recipients = new ArrayList<String>();
+        recipients.add(recipient);
+        return this.setToRecipients(recipients);
```
Comment (andrzej-majewski) [inline ikasaneip/builder/jar/src/main/java/org/ikasan/builder/component/endpoint/EmailProducerBuilderImpl.java]: why not to shortcut to

```
return this.setToRecipients(Arrays.asList(recipient))
```

File: ikasaneip/builder/jar/src/main/java/org/ikasan/builder/component/endpoint/EmailProducerBuilderImpl.java [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+        this.emailProducer.getConfiguration().setToRecipients(recipients);
+        return this;
+    }
+
+    @Override
+    public EmailProducerBuilder setCcRecipient(String recipient)
+    {
+        List<String> recipients = new ArrayList<String>();
+        recipients.add(recipient);
+        return this.setCcRecipients(recipients);
```
Comment (andrzej-majewski) [inline ikasaneip/builder/jar/src/main/java/org/ikasan/builder/component/endpoint/EmailProducerBuilderImpl.java]:
```
return this.setCcRecipients(Arrays.asList(recipient))
```

File: ikasaneip/builder/jar/src/main/java/org/ikasan/builder/component/endpoint/EmailProducerBuilderImpl.java [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+        this.emailProducer.getConfiguration().setCcRecipients(recipients);
+        return this;
+    }
+
+    @Override
+    public EmailProducerBuilder setBccRecipient(String recipient)
+    {
+        List<String> recipients = new ArrayList<String>();
+        recipients.add(recipient);
+        return this.setBccRecipients(recipients);
```
Comment (andrzej-majewski) [inline ikasaneip/builder/jar/src/main/java/org/ikasan/builder/component/endpoint/EmailProducerBuilderImpl.java]:
```
return this.setBccRecipients(Arrays.asList(recipient))
```

File: ikasaneip/builder/jar/src/main/java/org/ikasan/builder/component/endpoint/EmailProducerBuilderImpl.java (hunk 1 above)
Comment (mitcje) [reply ikasaneip/builder/jar/src/main/java/org/ikasan/builder/component/endpoint/EmailProducerBuilderImpl.java]: Updated

**Pull request merged**